doc/content/development/contributing.md

```diff
@@ -0,0 +1,127 @@
+
+## Referencing Issues
+
+In every pull request, we strongly recommend referencing an issue. This practice encourages robust
+discussions on whether the change is necessary and appropriate to Moltres. It also provides
+relevant context to the new changes proposed for quicker code review.
+
+## Work in a Fork
+
+Create your own [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) where you can
+make and commit your set of changes.
+
+### 1. Fork Moltres
+
+- Navigate to [https://github.com/arfc/moltres](https://github.com/arfc/moltres)
+- Click the "Fork" button near the upper right corner of the page.
+- Clone your new fork to your local machine using the following command (replace "username" with
+  your GitHub username):
+
+
+```bash
+git clone git@github.com:username/moltres.git
+```
+
+### 2. Add the `upstream` Remote
+
+Add the original Moltres repository as a remote named `upstream`:
+
+```bash
+cd moltres
+git remote add upstream git@github.com:arfc/moltres.git
+```
+
+### 3. Make Changes to Moltres
+
+Create a branch for your work:
+
+```bash
+git checkout -b branch_name
+```
+
+Make your changes and commit them to the branch you just created:
+
+```bash
+git add your_new_file.C your_new_file.h
+git commit -m "Your commit message here"
+```
+
+Ensure that your commit messages are insightful and descriptive such that they let people quickly
+understand your changes without having to read code. *Commit early and often* to create checkpoints
+that you can revert to if you break something while implementing your changes. Frequent commits
+also helps to keep your commits short and sweet.
+
+If you have been working on your changes for some time, there may have been new updates to Moltres
+implemented by others since the last time you pulled from the `upstream` remote. Merge those
+updates into your current branch by running the following commands while your current branch is
+checked out.
+
+```bash
+git fetch upstream
+git merge upstream/devel
+```
+
+!alert tip
+Refer to
+[Commit Often, Perfect Later, Publish Once: Git Best Practices](https://sethrobertson.github.io/GitBestPractices/)
+for an in-depth guide on Git workflows.
+
+### 4. Update Documentation
+
+If you make any changes to the source code which significantly alters functionality, please make
+the corresponding changes to the documentation in `moltres/doc/content/source` for the specific
+class you modified. You may build this website locally to review your documentation
+changes by running the following commands in the `moltres/doc` directory:
+
+```bash
+./moosedocs.py build --serve
+```
+
+!alert tip
+Feel free to reach out to anyone in the Moltres development team for help with documentation.
+
+!alert note
+Detailed documentation of Moltres syntax is currently a work in progress. However, feel free to
+contribute if you are familiar with the source code.
+
+### 5. Push Your Changes to Your GitHub Fork
+
+Push your branch to your fork on GitHub.
```

Comment:
```suggestion
Every commit you make is saved locally, but it is a good practice to have your remote repository synced with your local changes. You can do this by pushing your branch to your fork on GitHub.
```

Reply:
I agree with this statement about good practices, but in this scenario the main purpose of pushing is to upload the local changes in preparation for a pull request. I'll edit the text here to explain the purpose of pushing while also including your suggestion as a "best practices" tip.